### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by malsbat**
Specify how the transports are used, document all the APIs, and generate docs for the bindings.

### CODE REVIEW

File: src/topics.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ DPS_Status DPS_AddTopic(DPS_BitVector* bf, const char* topic, const char* separa
  * @param topic       The topic to match
  * @param separators  The separator strings for the topic
  *
- * @return - DPS_TRUE is there was match, DPS_FALSE is there was not a match.
+ * @return DPS_TRUE is there was match, DPS_FALSE is there was not a match.
```
Comment (georgen117) [inline src/topics.h]: DPS_TRUE **if** there was match, DPS_FALSE **if**

File: src/history.h [hunk 2]
```diff
@@ -33,47 +38,57 @@
...
 
+/**
+ * A list of node addresses and sequence numbers
+ */
 typedef struct _DPS_NodeAddressList {
-    uint32_t sn;
-    DPS_NodeAddress addr;
-    struct _DPS_NodeAddressList* next;
+    uint32_t sn;                /**< A sequence number */
+    DPS_NodeAddress addr;       /**< A node address */
```
Comment (georgen117) [inline src/history.h]: in some locations you add extra white space to the separate the documentation from the variables being documented in other locations you don't. I am fine with both but I would try and be consistent.

for example compare this documentation to the documentation done for  _DPS_KeyStore struct in dps.h where no white space was added.

From this single commit. It looks like typically whitespace is added to line up the documentation making reading the code and documentation a little easier.

File: src/topics.h (hunk 1 above)
Comment (malsbat) [reply src/topics.h]: Will fix

File: src/history.h (hunk 2 above)
Comment (malsbat) [reply src/history.h]: Agreed, but I'll do this in another pull request

**Pull request merged**